src/grid/mod.rs

```diff
@@ -5,9 +5,9 @@ use std::{fmt::Debug, io::BufRead};
 
 pub trait Grid: Debug {
     fn bands(&self) -> usize;
-    fn contains(&self, coord: &Coor4D) -> bool;
+    fn contains(&self, coord: &Coor4D, within: f64) -> bool;
     ///  Returns `None` if the grid or any of it's sub-grids do not contain the point.
-    fn interpolation(&self, coord: &Coor4D) -> Option<Coor4D>;
+    fn interpolation(&self, coord: &Coor4D, within: f64) -> Option<Coor4D>;
```

Comment:
@busstoptaktik worth some docs here describing what `within` should be?

Reply:
Just updated the trait with a (hopefully understandable) description